lib/portage/__init__.py

```diff
@@ -186,6 +187,7 @@
     sys.stderr.write(f"    {e}\n\n")
     raise
 
+utf8_mode = locale.getpreferredencoding(do_setlocale=False).lower() == "utf-8"
```

Comment:
Maybe `sys.getfilesystemencoding()` would be better here (also support Python UTF-8 Mode).